### ISSUE TITLE
quick question

### ISSUE THREAD

**Issue by ms888ekb** [liyunsheng13/BDL#48]
Hello,

In this scenario (mentioned in the previous post):

1) train CycleGAN to get translated images
(2) train BDL.py to get segmentation model with the translated images and source images,
(3) train SSL.py to get pseudo-labels and refine the segmentation model.
(4) retrain CycleGAN with an additional perceptual loss.

Do "train CycleGAN" mean one-batch iteration (A - B - A, B - A - B) or complete training phase with 20 epochs?

Thank you.

**Comment by liyunsheng13** [comment]
It means retraining cyclegan for 20 epochs.

**Comment by ms888ekb** [comment]
Thank you for the quick response!
And "train BDL.py" means the same - to train segmentation model through the full training dataset, not only a single batch, correct?

**Comment by liyunsheng13** [comment]
Yes

**Comment by ms888ekb** [comment]
Great, Thank you. I already implemented the whole forward pass (I'm working with geospatial data), now I'm confused on how to add perceptual loss to the CycleGAN model again. Once we've done the whole forward-direction training, we get:
- trained CycleGAN with 20 epochs
- trained adaptation segmentation model with several thousand steps + self-supervised iterations

from the article, I cannot still understand how CycleGAN can be further improved with the perceptual loss? What inputs should we pass to CycleGAN to improve it? I'm sorry if I explain too confusing, I try to understand how backwards direction works here. What we expect from the M model after the first forward pass to feed into the CycleGAN.
Thank you.

**Comment by ms888ekb** [comment]
Great, Thank you. I already implemented the whole forward pass (I'm working with geospatial data), now I'm thinking about how to add perceptual loss to the CycleGAN model again. Once we've done the whole forward-direction training, we get:

- trained CycleGAN with 20 epochs
- trained adaptation segmentation model with several thousand steps + self-supervised iterations

Do I understand right that to acquire the perceptual loss while the second CycleGAN training phase, we need:
Do one step of CycleGAN and infer S -> S'. Then pass S and S' to the already pre-trained adaptation segmentation M network, get the outputs and calculate the perceptual loss?

**Comment by liyunsheng13** [comment]
Yes, your understanding is correct. You can refer to the folder of cyclegan I uploaded to check the implementation detail.
